accounts/abi/bind/v2/base.go

```diff
@@ -89,25 +89,28 @@ type WatchOpts struct {
 
 // MetaData collects all metadata for a bound contract.
 type MetaData struct {
-	mu   sync.Mutex
-	Sigs map[string]string
-	Bin  string
-	ABI  string
-	ab   *abi.ABI
+	Sigs    map[string]string
+	Bin     string
+	ABI     string
+	Pattern string
+	Deps    []*MetaData
+
+	mu        sync.Mutex
+	parsedABI *abi.ABI
```

Comment:
If we wanted to instantiate the abi object the idiomatic way,  we could instantiate it using `sync.OnceValues`.  However, to do that we would have to introduce a constructor for `MetaData` (in order not to expose the instantation function returned from `sync.OnceValues` ).

Afaict, we do not accept instances of `MetaData` for any utility methods that abigen provides.  So modifying it does not incur backwards-compat issues?